### DISCUSSION TITLE
integration HA, MQTT and Zigbee2mqtt as 3 Docker-containers

### DISCUSSION THREAD

**Discussion by Ren656** [Koenkk/zigbee2mqtt#6075] [Devices]
hello, as a newbee I´m trying to find my way through in Homeassistant. I´m using Docker on a rpi4 (2g) and installed successfully three containers (Homeassistant, Mqtt and Zigbee2mqtt). However, HA does not show the two devices that were paired through the CC2531 connected thru usb with the rpi. The log.txt file does show the two devices. Apparently the installation of Zigbee2mqtt as Add-on in HA is missing but I don´t see how to install this 'third party add-on'  as this function seems not to be available in the docker-version of HA. What to do?
René

**Comment by m0wlheld** [comment]
You don‘t need any add-on, but you need to enable the MQTT integration in HA and make it connect to Mosquitto. 

I use a 5-container-setup (HA,Mosquitto,Zigbee2mqtt,MariaDB and Watchtower) on an RPi4B 4GB with docker-compose.

**Reply by Ren656** [reply]
@m0wlheld: Christoph, I have added the MQTT integration to HA (can be done in the App). I assume that this is what you meant by 'enable the MQTT integration´. Yet, I don´t see how to ´make it connect to Mosquitto´. Apparently you have installed ´Mosquitto' in a container where I have installed ´MQTT´, could that be the problem? To give a better picture I have included two screenshots: one of the set up in the ´docker-compose.yaml´, the other one the log.txt showing the actual situation.

Again, I still don´t see how to get the complete set up done. However, it could be very simple and I´m simply ´blind´.
René

/home/rene/Afbeeldingen/Schermafdruk van 2021-02-03 22-18-39.png
/home/rene/Afbeeldingen/Schermafdruk van 2021-02-04 16-57-09.png

**Reply by m0wlheld** [reply]
While adding MQTT integration to Home Assistant, there should have been a dialog to request MQTT server‘s details (host, username, password) from you.

**Reply by Ren656** [reply]
That´s right, i have put 192.168.2.6 and port 1883. For the time being I use no username and password.

**Reply by m0wlheld** [reply]
(Eclipse) Mosquitto is a MQTT server software. You need an MQTT server for Zigbee2mqtt. The MQTT integration of Home Assistant is a MQTT client. 

Of course, both Zigbee2mqtt and Home Assistant must connect to the same MQTT server. 

Btw: Did you enable the [homeassistant](https://www.zigbee2mqtt.io/information/configuration.html) discovery support in Zigbee2mqtt?

**Comment by Ren656** [comment]
Verstuurd vanaf mijn iPad

Op 4 feb. 2021 om 19:11 heeft Christoph Dahlen <notifications@github.com> het volgende geschreven:

﻿

(Eclipse) Mosquitto is a MQTT server software. You need an MQTT server for Zigbee2mqtt. The MQTT integration of Home Assistant is a MQTT client.

That’s how I have understood the whole concept of MQTT / mosquitto.

Of course, both Zigbee2mqtt and Home Assistant must connect to the same MQTT server.

They do

Btw: Did you enable the homeassistant<https://www.zigbee2mqtt.io/information/configuration.html> discovery support in Zigbee2mqtt?

In the ~/data/configuration.yalm I had set the first line to ‘homeassistant: true’ (contrary to the link you send). That one should be set to ‘false’?

**Reply by m0wlheld** [reply]
No, true is correct.

Sorry, but it should work that way, did you check the logs of both HA and Zigbee2MQTT?

**Comment by Ren656** [comment]
I checked the zigbee2mqtt-log, not the HA-one. Now I can see the following alerts (see below). If I have to do some interpretation: I think they are due to some experimenting with the available options from my side. Afterwards I was able to set the ‘listening’-parameter in MQTT to the topic ‘zigbee2mqtt/bridge/devices’ (quotation marks not included) resulting in a long list of the actual status of the two Ikea bulbs. Therefore I concluded that the installation of MQTT in HA was fine, but apparently not.

Unable to find referenced entities input_boolean.notify_alerts
3 februari 2021 22:35:32 – helpers/service.py (WARNING)
[2888961040] sequence item 0: expected str instance, Optional found
3 februari 2021 22:34:46 – Home Assistant WebSocket API (ERROR) - bericht kwam voor het eerst om 3 februari 2021 22:30:37 en verschijnt 4 keer
[2888961040] 'int' object is not iterable
3 februari 2021 22:34:04 – Home Assistant WebSocket API (ERROR)
[2888961040] MQTT topic name/filter must not be empty. for dictionary value @ data['topic']
3 februari 2021 22:27:13 – Home Assistant WebSocket API (ERROR)
Failed to connect to MQTT server due to exception: [Errno 111] Connection refused
3 februari 2021 8:49:42 – MQTT (ERROR)


Verstuurd vanaf mijn iPad

Op 4 feb. 2021 om 20:13 heeft Christoph Dahlen <notifications@github.com> het volgende geschreven:

﻿

No, true is correct.

Sorry, but it should work that way, did you check the logs of both HA and Zigbee2MQTT?

**Comment by Ren656** [comment]
What should I do with the Errno 111 connection error?

Verstuurd vanaf mijn iPad

Op 4 feb. 2021 om 20:13 heeft Christoph Dahlen <notifications@github.com> het volgende geschreven:

﻿

No, true is correct.

Sorry, but it should work that way, did you check the logs of both HA and Zigbee2MQTT?

**Reply by m0wlheld** [reply]
> What should I do with the Errno 111 connection error?

Fix it? Connection issue or missing/wrong authentication credentials.

**Reply by Ren656** [reply]
I read in the docker-compose yaml: both 'homeassistant' and 'zigbee2mqtt' contain a line with 'network_mode: host'. Is not the case with mqtt, there is the only specification for 'ports' (1883:1883 and 9001:9001). That is correct?

**Reply by m0wlheld** [reply]
It's an option. With `network_mode: host` a container can open any port for listening. For Home Assistant it is actually required, otherwise general device discovery using UPnP would not work.

Defining ports explicitely is an alternative, limiting the containers ability to open ports to the ones defines. MQTT uses port 1883 and 9001 only, so no need to use `network_mode: host` here.

Can you show your docker-compose file (without secrets) here? Please use the "Code" formatting (`<>` icon) of this editor.

**Comment by Ren656** [comment]
I´m linked to the rpi through Putty, resulting in a window with limited functions. I have been looking for a ´copy´-command but couldn´t find it. Does the attached screenshot work as well? The code is pretty straight forward without 'secrets´ in it.
![afbeelding](https://user-images.githubusercontent.com/71029393/106956759-5ca34c00-6737-11eb-94be-15b8414156bb.png)

**Reply by Ren656** [reply]
@m0wlheld, already thanks a lot for your help. I´ll find your answer tomorrow and 'call it a day'.

**Reply by m0wlheld** [reply]
Yeah, it‘s readable and looks okay. From a service perspective everything is properly set-up (and you use Mosquitto as well).

By default Mosquitto does not require authentication, so missing credentials in Home Assistant should not be an issue. 

Eventually I am out of ideas, especially since this:

> Afterwards I was able to set the ‘listening’-parameter in MQTT to the topic ‘zigbee2mqtt/bridge/devices’ (quotation marks not included) resulting in a long list of the actual status of the two Ikea bulbs.

can only be done with a working MQTT connection in HA. 

Is your Lovelace dashboard still in Auto-Mode, so discovered devices get automatically added to it?

**Comment by Ren656** [comment]
@m0wlheld: Christoph, I found the mistake. At least: I changed in the docker-compose.yaml the following line under homeassistant:
volumes:
  - /home/homeassistant/.homeassistant:/configuration.yaml (was: .../homeassistant:/config)

I had to redo the logging procedure in the HA app. After re-adding MQTT both ikea bulbs were found.

Again: thanks a lot for your support!
